compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -739,6 +739,24 @@ object Types {
       else ctx.asSeenFrom(this, pre, cls)
     }
 
+    /** The JVM tag for this type, if it's a primitive, `java.lang.Object` otherwise. */
+    final def toTag(implicit ctx: Context): Name = {
+      classSymbol match {
+        case t if t eq defn.IntClass     => nme.specializedTypeNames.Int
+        case t if t eq defn.BooleanClass => nme.specializedTypeNames.Boolean
+        case t if t eq defn.ByteClass    => nme.specializedTypeNames.Byte
+        case t if t eq defn.LongClass    => nme.specializedTypeNames.Long
+        case t if t eq defn.ShortClass   => nme.specializedTypeNames.Short
+        case t if t eq defn.FloatClass   => nme.specializedTypeNames.Float
+        case t if t eq defn.UnitClass    => nme.specializedTypeNames.Void
+        case t if t eq defn.DoubleClass  => nme.specializedTypeNames.Double
+        case t if t eq defn.CharClass    => nme.specializedTypeNames.Char
+        case _                           => nme.specializedTypeNames.Object
+      }
+    }
+
```

Comment:
This should probably go in `Definitions`, where we define functionality related to numeric classes. `Types` is not concerned with these. I suggest to  make it another parameter of `Definitions.valueTypeRef`.

Reply:
Addressed in https://github.com/lampepfl/dotty/pull/3234/commits/fb6e1f1f75b756f68a109369c622fb5a6a7956a5. Is that what you had in mind?